code/addons/controls/template/stories/sorting.stories.ts

```diff
@@ -31,3 +31,14 @@ export const None = { parameters: { controls: { sort: 'none' } } };
 export const Alpha = { parameters: { controls: { sort: 'alpha' } } };
 
 export const RequiredFirst = { parameters: { controls: { sort: 'requiredFirst' } } };
+
+export const CustomSort = {
+  parameters: {
+    controls: {
+      sort: (a, b) =>
+        a.table?.category?.localeCompare(b.table?.category) ||
+        (a.type?.required === b.type?.required ? 0 : a.type?.required ? 1 : -1) ||
+        0,
+    },
```

Comment:
**question (blocking):** @kylegach @jonniebigodes just asking you both because GitHub suggests you both
<img width="416" alt="image" src="https://github.com/user-attachments/assets/bef6dce8-5f96-4673-8e0f-487fd0e0e699">

I tested to pass a callback function here, however [`const { expanded, sort, presetColors } = useParameter<ControlsParameters>(PARAM_KEY, {});`](https://github.com/Shinigami92/storybook/blob/8a18f23e9590f8d465f7947c2f6cf61c59b4fa30/code/addons/controls/src/ControlsPanel.tsx#L54) then returns `undefined` for `sort`.
I also tried to pass e.g. `'custom'` as string, and then it successfully pass through as is. My assumption is that somewhere **runtime-functions are not serializable** and results in `undefined`.
Is someone knowing what that could be and point me in the right direction?

Reply:
@Shinigami92 parameters are loaded into Storybook's "preview" (the inner iframe that renders the stories). They are then serialized across a channel into the "manager" (the outer iframe that's Storybook's actual UI) using a library called `telejson`.

Historically, telejson serialized pure functions by default. But then we started getting complaints about the use of `eval`, which is forbidden in some companies. Now function serialization is opt-in using [channelOptions.allowFunction](https://storybook.js.org/docs/api/main-config/main-config-core#channeloptionsallowfunction). However, I think we might even remove that in SB9 because the `eval` code still exists in `telejson` which triggers some security checks.

If you want a function to be available in the server, your best bet is to have users set it in `.storybook/manager.js`, see the [`renderLabel` example](https://storybook.js.org/docs/configure/user-interface/sidebar-and-urls#auto-title-filename-case) (which strangely doesn't seem to be documented anywhere else @kylegach @jonniebigodes). The manager APIs are sorely underdeveloped (something that @ndelangen wants to do something about, but which is currently lower priority) and in your case I think you really want the function to be available in both the manager AND the preview and we really don't have a great solution for that.

I'm open to suggestions for better ways to approach this -- it's definitely an area that could use work. We're talking about some architectural changes next year that could solve this in a very different way. But nothing in the short term.

Reply:
@shilman OH 😲 this makes a lot of sense!
As you said, I could try to allow to register a function in manager and then just pass the registered function-name as string. And then see if that can be used.
If not, then I might need to wait until SB-9.

Reply:
@ndelangen another good use case for the annotation server